### PR TITLE
Add Wasi Workload Manifest

### PR DESCRIPTION
**Pull request by steveisok**

### CODE REVIEW

**Comment by steveisok** [comment]
Official build works.  I think we'll need to add in build signing similar to emsdk next.

https://dev.azure.com/dnceng/internal/_build/results?buildId=2169069&view=results

File: eng/Publishing.props [hunk 1]
```diff
@@ -2,5 +2,45 @@
 <Project>
    <PropertyGroup>
       <PublishingVersion>3</PublishingVersion>
+      <AutoGenerateSymbolPackages>false</AutoGenerateSymbolPackages> <!-- we don't need symbol packages for emsdk -->
```
Comment (directhex) [inline eng/Publishing.props]: symbols packages for wasi, not emsdk

File: eng/Versions.props [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
 <Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
+    <MajorVersion>8</MajorVersion>
+    <MinorVersion>0</MinorVersion>
     <VersionPrefix>19.0.0</VersionPrefix>
-    <PreReleaseVersionLabel>alpha</PreReleaseVersionLabel>
-    <PreReleaseVersionIteration>1</PreReleaseVersionIteration>
+    <PreReleaseVersionLabel>preview</PreReleaseVersionLabel>
+    <PreReleaseVersionIteration>4</PreReleaseVersionIteration>
+    <WasiVersion>19.0.0</WasiVersion>
```
Comment (directhex) [inline eng/Versions.props]: We should either use VersionPrefix in lieu of WasiVersion, or make `<WasiVersion>$(VersionPrefix)</WasiVersion>`. No point having the same info twice

File: eng/nuget/Microsoft.NET.Workload.Wasi.Current.Manifest/WorkloadManifest.json.in [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+  },
+  "packs": {
+    "Microsoft.NET.Runtime.Wasi.Sdk" : {
+      "kind": "Sdk",
+      "version": "${PackageVersion}",
+      "alias-to": {
+        "win-x64": "Microsoft.NET.Runtime.Wasi.${WasiVersion}.Sdk.win-x64",
+        "win-arm64": "Microsoft.NET.Runtime.Wasi.${WasiVersion}.Sdk.win-x64",
+        "linux-x64": "Microsoft.NET.Runtime.Wasi.${WasiVersion}.Sdk.linux-x64",
```
Comment (directhex) [inline eng/nuget/Microsoft.NET.Workload.Wasi.Current.Manifest/WorkloadManifest.json.in:20]: linux-arm64 entry needed. We have linux-arm64 builds

File: eng/nuget/Microsoft.NET.Workload.Wasi.Current.Manifest/WorkloadManifest.json.in [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "version": "${WorkloadVersion}",
+  "workloads": {
+    "microsoft-net-sdk-wasi": {
+      "abstract": true,
+      "description": "WASI SDK compiler tooling",
+      "packs": [
+        "Microsoft.NET.Runtime.Wasi.Sdk"
+      ],
+      "platforms": [ "win-x64", "win-arm64", "linux-x64", "osx-x64", "osx-arm64" ]
```
Comment (directhex) [inline eng/nuget/Microsoft.NET.Workload.Wasi.Current.Manifest/WorkloadManifest.json.in]: linux-arm64 here too

File: eng/nuget/Microsoft.NET.Workload.Wasi.Current.Manifest/WorkloadManifest.targets [hunk 5]
```diff
@@ -0,0 +1,33 @@
+<Project>
+  <ImportGroup Condition="'$(RuntimeIdentifier)' == 'wasi-wasm' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals('$(TargetFrameworkVersion)', '8.0'))">
+    <Import Project="Sdk.props" Sdk="Microsoft.NET.Runtime.Wasi.Sdk" />
+  </ImportGroup>
+  <!--
```
Comment (directhex) [inline eng/nuget/Microsoft.NET.Workload.Wasi.Current.Manifest/WorkloadManifest.targets]: This whole section is commented out. Delete it?

File: eng/Publishing.props [hunk 6]
```diff
@@ -2,5 +2,45 @@
...
+  <PropertyGroup>
+    <_UploadPathRoot>wasi-sdk</_UploadPathRoot>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <PublishDependsOnTargets>$(PublishDependsOnTargets);_PublishInstallers</PublishDependsOnTargets>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <_InstallersToPublish Include="$(ArtifactsDir)**\*.wixpack.zip" />
```
Comment (akoeplinger) [inline eng/Publishing.props]: probably want to use `$(ArtifactsDir)\packages` or wherever this is produced so we don't accidentally grab files from obj

**Review by akoeplinger — APPROVED** [review]

File: eng/nuget/Microsoft.NET.Workload.Wasi.Current.Manifest/WorkloadManifest.json.in [hunk 7]
```diff
@@ -16,8 +16,9 @@
       "version": "${PackageVersion}",
       "alias-to": {
         "win-x64": "Microsoft.NET.Runtime.Wasi.${WasiVersion}.Sdk.win-x64",
-        "win-arm64": "Microsoft.NET.Runtime.Wasi.${WasiVersion}.Sdk.win-x64",
+        "win-arm64": "Microsoft.NET.Runtime.Wasi.${WasiVersion}.Sdk.win-arm64",
         "linux-x64": "Microsoft.NET.Runtime.Wasi.${WasiVersion}.Sdk.linux-x64",
+        "linux-arm64": "Microsoft.NET.Runtime.Wasi.${WasiVersion}.Sdk.linux-arm64",
         "osx-x64": "Microsoft.NET.Runtime.Wasi.${WasiVersion}.Sdk.osx-x64",
         "osx-arm64": "Microsoft.NET.Runtime.Wasi.${WasiVersion}.Sdk.osx-x64"
```
Comment (directhex) [inline eng/nuget/Microsoft.NET.Workload.Wasi.Current.Manifest/WorkloadManifest.json.in]: osx-arm64

**Review by directhex — APPROVED** [review]

**Pull request merged**